pkg/serverconfig/config.go

```diff
@@ -253,6 +253,13 @@ func addCustomization(fs *flag.FlagSet, customization *Customization) {
 			klog.Fatalf("Could not marshal ConsoleConfig customization.developerCatalog.categories field: %v", err)
 		}
 	}
+
+	addPage, err := json.Marshal(customization.AddPage)
```

Comment:
Is this needed if we're already validating the flag set in validate.go?

Reply:
This is required to parse customization from the YAML file, and override the fs value in the if below.

The `validate.go` checks if the bridge was called with `-add-page-json` and an invalid JSON.

Reply:
Ack. Removed `JSON` and `-json` suffixes

Reply:
To be clear, this is required to convert the YAML to JSON? It's worth a comment here explaining why we're parsing and then stringifying.

Reply:
Sorry I doesn't explained this well.

The complete "add-page" customization object is saved internally as a JSON string in this fs object. This is similar to the existing customization and make it easier to save them in main.go and read by the indexHandler.

This marshal here converts this data from the config.yaml to the JSON. So indexHandler then can just render this string.

The additional check in `validate.go` is only required for the case when the user provides this data as command line argument. `bin/bridge -config .... -add-page "{}"` This JSON could be broken or can contain unknown keys.